Automattic/jetpack
# Beta plugin: Try using core's ajax updater for applying updates

#### Changes proposed in this Pull Request:

They didn't do all that great of a job of making it generally usable,
but it seems to work ok if we avoid some of the higher-level functions.

#### Jetpack product discussion


p1626881378211900/1626879820.209200-slack-CBG1CP4EN

#### Does this pull request change what data or activity we track or use?


No.

#### Testing instructions:






* Use the Beta plugin to install Jetpack 9.9.
* Go to the plugin select list and use the notice to update to 9.9.1. The notice should disappear without reloading the page, and the line for Jetpack should reflect version 9.9.1.

----

* Use the Beta plugin to install Jetpack 9.9, then activate master.
* Go to the plugin select list and use the notice to update 9.9 to 9.9.1. The notice should disappear without reloading the page, and the line for Jetpack should still reflect "Bleeding Edge".

----

* Use the Beta plugin to install Jetpack 9.9, then deactivate Jetpack.
* Go to the plugin select list and use the notice to update to 9.9.1. The notice should disappear without reloading the page, and the line for Jetpack should still reflect Jetpack not being active.

----

* Use the Beta plugin to install Jetpack 9.9, then activate master.
* Edit `/path/to/wp-content/plugins/jetpack-dev/.jpbeta.json` to change the "version" field to "10.0-alpha-1".
* Go to the plugin select list and use the notice to update "Bleeding Edge". The notice should disappear without reloading the page, and the line for Jetpack should reflect the new version.

----

* Use the Beta plugin to install Jetpack 9.9.
* Go to the plugin management page for Jetpack and use the notice to update to 9.9.1. The notice should disappear without reloading the page, the "Currently Running" box should reflect version 9.9.1, and "Latest Stable" should be shown as active.

----

* Use the Beta plugin to install Jetpack 9.9, then activate master.
* Go to the plugin management page for Jetpack and use the notice to update to 9.9.1. The notice should disappear without reloading the page, the "Currently Running" box should still reflect Bleeding Edge, and "Bleeding Edge" should still be shown as active.

----

* Use the Beta plugin to install Jetpack 9.9, then deactivate Jetpack.
* Go to the plugin management page for Jetpack and use the notice to update to 9.9.1. The notice should disappear without reloading the page, but there should still be no "Currently Running" box and no version should be shown as active.

----

* Use the Beta plugin to install Jetpack 9.9, then activate master.
* Edit `/path/to/wp-content/plugins/jetpack-dev/.jpbeta.json` to change the "version" field to "10.0-alpha-1".
* Go to the plugin management page for Jetpack and use the notice to update "Bleeding Edge". The notice should disappear without reloading the page, the "Currently Running" box should reflect the new version, and "Bleeding Edge" should be shown as active.

----

* Use the Beta plugin to install Jetpack 9.8.
* Edit `/path/to/wp-content/plugins/jetpack/jetpack.php` to change the "Version" field on line 7 to "9.8-alpha".
* Go to the plugin management page for Jetpack. Note there's an "Existing Version" box shown for "9.8-alpha".
* Use the notice to update to 9.9.1. The notice should disappear without reloading the page, the "Currently Running" box should reflect the new version, the "Currently Running" box should reflect version 9.9.1, "Latest Stable" should be shown as active, and the "Existing Version" box should have disappeared.

----

* Use the Beta plugin to install Jetpack 9.9, then activate master.
* Edit `/path/to/wp-content/plugins/jetpack/jetpack.php` to change the "Version" field on line 7 to "9.8-alpha".
* Edit `/path/to/wp-content/plugins/jetpack-dev/.jpbeta.json` to change the "version" field to "10.0-alpha-1".
* Go to the plugin management page for Jetpack. Note there's an "Existing Version" box shown for "9.8-alpha".
* Use the notice to update 9.8-alpha to 9.9.1. The notice should disappear without reloading the page, the "Currently Running" box should still reflect "10.0-alpha-1", "Bleeding Edge" should still be shown as active, and the "Existing Version" box should have disappeared.

----

* Use the Beta plugin to install Jetpack 9.9, then activate master.
* Edit `/path/to/wp-content/plugins/jetpack/jetpack.php` to change the "Version" field on line 7 to "9.8-alpha".
* Edit `/path/to/wp-content/plugins/jetpack-dev/.jpbeta.json` to change the "version" field to "10.0-alpha-1".
* Go to the plugin management page for Jetpack. Note there's an "Existing Version" box shown for "9.8-alpha".
* Use the notice to update "Bleeding Edge". The notice should disappear without reloading the page, the "Currently Running" box should reflect the new version, "Bleeding Edge" should still be shown as active, and the "Existing Version" box should still be present.
